MVP blog in Docusaurus

laurakalbag opened the pull request:
Preview at: https://docs-git-laurakalbag-sta-3518-spike-blog-in-do-5a3a3c-statelyai.vercel.app/blog

This spike takes the blog posts from stately.ai/blog (in the blog repo) and puts them into the Docusaurus blog in the same platform as the docs. This would make the blog and docs manageable under one platform, make it easy to cross-link/embed, and any dev will benefit both the docs and the blog.

As you can tell from the commits, most of the effort was wrangling slugs and relative URLs. Note for future Laura: if in doubt: give the whole path to the file, including the file extension (e.g: `/blog/2023-02-06-github-import-machines/index.mdx`) and let Docusaurus resolve it correctly.

There’s a few [issues still yet to be addressed on Linear](https://linear.app/statelyai/project/advocacydocs-mvp-blog-in-docusaurus-1523e9a9f106), these are mostly some polish required on the templates and styles. ~~The big question is whether it’s manageable to deliver the docs at /docs and the blog at /blog from one platform…~~ @mellson says yes 😁 

**TODO**

- [x] Fix pricing link
- [x] Fix arrows on landing page
- [x] Fix heading styles (just `h2`s?) on blog posts (LK)
- [x] Ensure videos/embeds are displayed as blocks, not inline (LK)
- [x] Look into how Plausible should be embedded on the landing page (if at all) (( note, I think it should be included in the static file served during ssr at `static/index.html` but not the one we load during csr `static/landing-page/index.html` ))

mellson commented:
> This spike takes the blog posts from stately.ai/blog (in the blog repo) and puts them into the Docusaurus blog in the same platform as the docs. This would make the blog and docs manageable under one platform, make it easy to cross-link/embed, and any dev will benefit both the docs and the blog.
> 
> As you can tell from the commits, most of the effort was wrangling slugs and relative URLs. Note for future Laura: if in doubt: give the whole path to the file, including the file extension (e.g: `/blog/2023-02-06-github-import-machines/index.mdx`) and let Docusaurus resolve it correctly.
> 
> There’s a few [issues still yet to be addressed listed as subtasks on Linear](https://linear.app/statelyai/issue/STA-3518), these are mostly some polish required on the templates and styles. The big question is whether it’s manageable to deliver the docs at /docs and the blog at /blog from one platform… @mellson (please don’t hate me) 😬

Great work @laurakalbag 👏🏻 

The easiest technical option here is to move the entire landing page to Docusaurus.
Then we don't need our custom landing page or routing setup; we can instead opt into all the Docusaurus defaults like having docs at /docs and blog at /blog.

laurakalbag commented:
@mellson Fixed a lot of the layout issues on the landing page. Here’s what I worked out so far 🕵️ 
- For some reason the `<source>` alternatives inside `<picture>` were being rendered as invisible blocks. This is what was causing the weird spacing on the testimonials (the avatars were essentially double height). For now I just removed the extra `<source>`s in the `<picture>`s and their corresponding `webp` images. They were a nice-to-have.
- Docusaurus default styles for headings and paragraphs were being applied (because the landing page was relying on browser defaults for those.) So overriding those worked where necessary and fixed some of the odd sized and spaced headings.

That second fix improved a lot of the issues with arrow positioning. I suspect the remaining not-quite-aligned arrows are caused by similar Docusaurus styling, but I can’t quite get to the bottom of them right now!

mellson reviewed (approved):
LGTM. It's a big change, so we should expect some follow-up fixes. But it's looking really good so far. Great work @laurakalbag 👏🏻 👏🏻 👏🏻
